VatSpace/cats.lua

```diff
@@ -0,0 +1,83 @@
+-- cat module
+local sounds = require( "sounds" )
+local ui = require( "ui" )
+local gv = require( "gamevariables" )
+
+local M = {}
+
+M.cat = nil
+
+local function moveCat(cat)
+    cat.x = cat.x - cat.speed
+end
+
+local function showExtraTime(x, y)
+    local time = display.newText("+10s", x, y, "munro.ttf", 15)
+    timer.performWithDelay(35, function(event)
+        time.y = time.y - 1
+        if event.count == 10 then
+            time:removeSelf()
+            time=nil
+        end
+    end, 10)
+end
+
+local function tapCatInSpace(event)
+    if ( event.phase == "ended" ) then
+        cat = event.target
+        if (cat.hasTime) then
+            sounds.playCatMeow()
+            showExtraTime(cat.x + 10, cat.y - 25)
+            gv.time = gv.time + 10
+            cat.hasTime = false
+        end
+    end
+    return true
+end
+
+local function randomizeFieldsFor(cat)
+    cat.isVisible = false
+    visibilty = {false, false, true}
+    x = math.random(1, 3)
+    cat.isVisible = visibilty[x]
```

Comment:
its not necessary to create table with false,false,true etc. what if we want for example 0.03 chance for something?
probably the best solution is use ternary operator:
cat.isVisible = math.random(1,3) > 2 and true or false
and maybe that is too big chance for cat? but I didn't see this feature in action so I leave it to you 😸

Reply:
Thanks! That's great idea (however of course I would'n use a table to create 0.03 chance for something :P )  I did few manual test and it's like with "random", you never know, sometimes there is 3 cats in row and sometimes 30 sec without cat. I recommend to leave it like this for now.